dd-sdk-android/src/test/kotlin/com/datadog/android/log/internal/logger/LogcatLogHandlerTest.kt

```diff
@@ -84,4 +86,57 @@ internal class LogcatLogHandlerTest {
                     "\$runnable\$1"
             )
     }
+
+    @Test
+    fun `resolves valid stack trace element when wrapped in timber`(forge: Forge) {
+
+        // Given
+        val forgeFileName: Forge.() -> String = {
+            "${this.anAlphabeticalString(Case.ANY)}.${this.aStringMatching("(kt|java)")}"
+        }
+
+        val ignoredElements = forge.aList {
+
+            val className = if (aBool()) {
+                // generate from ignored class names pattern
+                LogcatLogHandler.IGNORED_CLASS_NAMES.random()
+            } else {
+                // generate from ignored packages prefixes pattern
+                val packagePrefix = LogcatLogHandler.IGNORED_PACKAGE_PREFIXES.random()
+                packagePrefix + ".${anAlphabeticalString(Case.ANY).capitalize()}"
+            }
+
+            StackTraceElement(
+                className,
+                anAlphabeticalString(Case.ANY),
+                forgeFileName(this),
+                aSmallInt()
+            )
+        }
+
+        val validElements = forge.aList {
+
+            val className = "com.${anAlphabeticalString(Case.LOWER, 5)}" +
```

Comment:
Is there any chance that some of these packages can collide with the ignored ones ?

Reply:
no, there is no such entry matching `com.{5 chars}.{6 chars}.{7 chars}.{8 chars}`. It can be introduced at some point in future, but the probability of that is very low I guess. I just didn't want to introduce something more complex for that generation.